tests/test.sh

```diff
@@ -12,31 +14,36 @@ setup-data-libraries --help
 get-tool-list --help
 
 echo "Starting galaxy docker container"
-CID=`docker run -d -e GALAXY_CONFIG_WATCH_TOOL_DATA_DIR=True -p 8080:80 bgruening/galaxy-stable`
+# We start the image with the -P flag that published all exposed container ports
+# to random free ports on the host, since on OS X the container can't be reached
+# through the internal network (https://docs.docker.com/docker-for-mac/networking/#i-cannot-ping-my-containers)
+CID=`docker run -d -e GALAXY_CONFIG_WATCH_TOOL_DATA_DIR=True -P bgruening/galaxy-stable`
```

Comment:
I guess the disadvantage is that now all docker container ports are exposed instead of only the web port. Does that really matter though, in terms of security? At least port collision is not a concern any more.

Reply:
It isn't more than it was before, given that the web port is reachable with a predictable api key / username/password combination, but that's certainly something we should change.